### PR TITLE
[HDInsight]Add New API Version 2023-08-15-Preview

### PR DESCRIPTION
**Pull request by xinlifoobar**
…view

# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

[1] [public repo review queue], [private repo review queue]  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.  
[2] https://aka.ms/azsdk/support/specreview-channel  
[3] [List of SDK breaking changes approvers] in pinned Teams announcement  
[4] [public repo merge queue], [private repo merge queue]  

If you need further help with anything, see `Getting help` section below.

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to Step 2, "ARM Review", for this PR.

### Breaking changes review (Step 1)

- If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
  - The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
  - You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

### ARM API changes review (Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

## Getting help

- For general PR approval workflow, see the diagram at the top of this comment.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[public repo review queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AWaitForARMFeedback+-label%3AIDCDevDiv+draft%3Afalse+sort%3Acreated-asc
[private repo review queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AWaitForARMFeedback+-label%3AIDCDevDiv+draft%3Afalse+sort%3Acreated-asc
[List of SDK breaking changes approvers]: https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1689115217750?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1689115217750&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1689115217750
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AIDCDevDiv+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AIDCDevDiv+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Comment by xinlifoobar** [comment]
Hi @rkmanda. This is our preview API version and is not public yet. Could you please help us merge it?

**Comment by maririos** [comment]
> Hi @rkmanda. This is our preview API version and is not public yet. Could you please help us merge it?

You need to follow the process described on https://github.com/Azure/azure-rest-api-specs/pull/25135#issuecomment-1663197460 to get an approval on the detection of the breaking changes

**Comment by TimLovellSmith** [comment]
Is it OK if I fix the purpose of the PR to 'add a new api version' since its now a new api version?

>Purpose of this PR
What's the purpose of this PR? Check all that apply. This is mandatory!

 New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
 Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
 Update existing version to fix swagger quality issues in S360.
 Other, please clarify:
edit this with your clarification

**Comment by TimLovellSmith** [comment]
I *think* all these avocado failures probably need to be fixed to get this PR properly linted.

>The swagger JSON file is not referenced from the readme file.
readme: [specification/hdinsight/resource-manager/readme.md](https://github.com/Azure/azure-rest-api-specs/blob/b4f7b0b3b062c419843da7d990d564ac68b47284/specification/hdinsight/resource-manager/readme.md)
json: [Microsoft.HDInsight/preview/2023-08-15-preview/applications.json](https://github.com/Azure/azure-rest-api-specs/blob/b4f7b0b3b062c419843da7d990d564ac68b47284/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2023-08-15-preview/applications.json)

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2023-04-15-preview/cluster.json [hunk 1]
```diff
@@ -1237,6 +1237,18 @@
       "type": "object",
       "description": "The network properties.",
       "properties": {
+        "outboundDependenciesManagedType": {
```
Comment (TimLovellSmith) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2023-04-15-preview/cluster.json]: >outboundDependenciesManagedType [](http://example.com/codeflow?start=9&length=31)

I feel like I'm missing some context for this documentation to make sense. What kind of outbound dependencies is this talking about, and what kind of managing is done to them? Anthing you could add to clarify?

**Comment by TimLovellSmith** [comment]
Which commits can I diff to see just the changes introduced in the new apiversion? If its hard to explain, it might even be worth redoing all your commits using openAPIHub and force pushing those commits over this branch, it should give us something easy to review. Or just adding commits that revert and then reapply the differences to the old api version.

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2023-04-15-preview/cluster.json (hunk 1 above)
Comment (xinlifoobar) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2023-04-15-preview/cluster.json]: Let me add one document here to explain the change. [HDInsight Cluster Spec Change Proposal] https://microsoftapc-my.sharepoint.com/:w:/g/personal/xinli1_microsoft_com/EaQq56jg_WJCr1-jlrU1hRQBVPl09ltZ5sCI6hdS-S9AfQ?e=IZ1zrJ

HDInsight provides the experience of Hadoop cluster as a service. Internally they are a set of VMs and their virtual network configurations. Previous, we let the customer bring & manage their own network solutions including NSG, Firewall, and 3rd party integrations while today some Cx require us to manage. That's how we define 'Managed' and 'External'

**Comment by xinlifoobar** [comment]
> Which commits can I diff to see just the changes introduced in the new apiversion? If its hard to explain, it might even be worth redoing all your commits using openAPIHub and force pushing those commits over this branch, it should give us something easy to review. Or just adding commits that revert and then reapply the differences to the old api version.

**Comment by xinlifoobar** [comment]
> Which commits can I diff to see just the changes introduced in the new apiversion? If its hard to explain, it might even be worth redoing all your commits using openAPIHub and force pushing those commits over this branch, it should give us something easy to review. Or just adding commits that revert and then reapply the differences to the old api version.

I tried OpenAPIHub but failed with below error so have to do this manually. I combined all in the same commit and hopefully this works. Thanks.

```
URL: https://openapihub.azure-devex-tools.com/branch/create

ApiError: Server encountered an internal server error.
    During API SAVE_NEW_SWAGGER_VERSION({"baseBranch":"main","isNewService":false,"branchOptions":"UnprotectedBranch","isNewDataPlane":false,"baseBranchType":"Default","confidentiality":"public","serviceName":"hdinsight","serviceType":"resource-manager","hasNewDataPlane":false,"rpName":"Microsoft.HDInsight","releaseState":"preview","version":"2023-08-15-preview","copyExamples":true,"basedOnVersion":"preview/2023-04-15-preview"})
    at https://openapihub.azure-devex-tools.com/static/build.js:415:147161
    at Generator.next (<anonymous>)
    at a (https://openapihub.azure-devex-tools.com/static/build.js:415:146160)
The above error occurred in task SAVE_NEW_SWAGGER_VERSION
    created by CREATE_NEW/SAVE_NEW_SWAGGER_VERSION
    created by anonymous

```

**Comment by xinlifoobar** [comment]
> 



> I _think_ all these avocado failures probably need to be fixed to get this PR properly linted.
> 
> > The swagger JSON file is not referenced from the readme file.
> > readme: [specification/hdinsight/resource-manager/readme.md](https://github.com/Azure/azure-rest-api-specs/blob/b4f7b0b3b062c419843da7d990d564ac68b47284/specification/hdinsight/resource-manager/readme.md)
> > json: [Microsoft.HDInsight/preview/2023-08-15-preview/applications.json](https://github.com/Azure/azure-rest-api-specs/blob/b4f7b0b3b062c419843da7d990d564ac68b47284/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2023-08-15-preview/applications.json)

Done.

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2023-04-15-preview/cluster.json [hunk 2]
```diff
@@ -1237,6 +1237,18 @@
...
+          "type": "string",
+          "description": "A value indicating how that the outbound dependencies will be managed",
```
[the remark this answers is not in the document]
Comment (xinlifoobar) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2023-04-15-preview/cluster.json]: Great feedback. Updated!

**Comment by ms-henglu** [comment]
Hi @xinlifoobar, I'd like to double confirm that the only change introduced between 2023-04-15-preview and 2023-08-15-preview is "outboundDependenciesManagedType"?

**Comment by xinlifoobar** [comment]
> Hi @xinlifoobar, I'd like to double confirm that the only change introduced between 2023-04-15-preview and 2023-08-15-preview is "outboundDependenciesManagedType"?

I have to do this because this saves effort than bypassing the breaking change process. Also, there are a lot more works behind this enum value so it is better to be in a new api version.

**Review by ms-henglu — APPROVED** [review]

**Pull request merged**